primerj/src/main/java/org/primer/primerj/core/TxBuilder.java

```diff
@@ -621,9 +627,8 @@ public int compare(Out out1, Out out2) {
         while (true) {
             long fees = 0;
 
-            if (lastCalculatedSize >= 1000) {
-                // If the size is exactly 1000 bytes then we'll over-pay, but this should be rare.
-                fees += (lastCalculatedSize / 1000 + 1) * Utils.getFeeBase();
+            if (lastCalculatedSize >= 0) {
+                fees += lastCalculatedSize * (Utils.getFeeBase() / 1000);
             }
             if (needAtLeastReferenceFee && fees < Utils.getFeeBase())
                 fees = Utils.getFeeBase();
```

Comment:
this if (needAtLeastReferenceFee) {} probably should be removed too

Reply:
line 648~662, the code block if (!needAtLeastReferenceFee) {}
i think also need to be deleted as primecoin does not waive transaction fee like bitcoin

Reply:
line 705: 
if (size / 1000 > lastCalculatedSize / 1000 && Utils.getFeeBase() > 0)
this condition may need to be updated as fee needs to go up once the estimated size goes up, it's no longer a KB step-up.